.githooks/pre-commit

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -e
+
+# Optional Pre-Commit Hook
+echo "Running optional pre-commit hook..."
+just lint
+just lint-specs-spelling-check
```

Comment:
There are a lot of options here, including automatically amending the commit with the linting fixes. Here, we make the commit and our working directory is updated with the fixes, allowing the user to make another commit to fix the problems (or amend the previous one if they want). 

Spelling will not automatically be fixed, but prevents the commit if there are spelling mistakes.